current/runtime/fees.md

```diff
@@ -4,118 +4,151 @@ lang: en
 title: Transaction Fees
 ---
 
-When transactions are submitted to a blockchain, they are executed by the nodes in the network. To
-be economically sustainable, nodes charge a fee to execute a transaction. This fee must be covered
-by the sender of the transaction. The cost to execute transactions can vary, so Substrate provides a
-flexible mechanism to characterize the minimum cost to include a transaction in a block.
+When transactions (extrinsics) are submitted to a blockchain, they are executed (dispatched) by the
+nodes in the network. To be economically sustainable, nodes charge a fee to dispatch a transaction.
+This fee must be covered by the sender of the transaction. The cost to dispatch transactions can
+vary, so Substrate provides a flexible mechanism called "weights" to characterize the minimum cost to
+include a transaction in a block.
 
 The fee system is heavily linked to the [weight system](/current/learn-substrate/weight.md). Make 
 sure to understand weights before reading this document.
 
-## Default Fees
+## Inclusion Fee
 
-The default fee system uses the uses _fixed_ weights represented by the following enum:
+A transaction fee consists of three parts:
+
+* `base_fee`: A fixed fee that is applied to every transaction. See
+  [`TransactionBaseFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionBaseFee).
+* `length_fee`: A per-byte fee that is multiplied by the number of bytes in the encoded transaction. See
+  [`TransactionByteFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionByteFee).
+* `weight_fee`: A per-weight-unit fee that is multiplied by the weight of the transaction. The
+  weight of each dispatch is denoted via the flexible `#[weight]` annotation. The weight must be
+  converted to the `Currency` type. For this, each runtime must define a
+  [`WeightToFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.WeightToFee)
+  type that makes the conversion. `WeightToFee` must be a struct that implements [`Convert<Weight,
+  Balance>`](https://substrate.dev/rustdocs/master/sp_runtime/traits/trait.Convert.html).
+
+The sum of the three transaction fee components (i.e. `base_fee + length_fee + weight_fee`) is known
+as the _inclusion fee_. Even if a transaction fails, the signer must pay the inclusion fee.
```

Comment:
Sort of, this can be more precise.

1. The fee is deducted _pre-dispatch,_ so there really is no concept of "succeeded" or "failed". This fee is gone before transaction execution even begins.
2. The Transaction Payment pallet requires that the account stay alive, i.e. not drop below the existential deposit due to the transaction fee. In that case, it would _try_ to pay the fee but fail and never even start execution.